cpp/src/search/contains.cu

```diff
@@ -51,20 +69,28 @@ struct contains_scalar_dispatch {
     using DType           = device_storage_type_t<Type>;
     using ScalarType      = cudf::scalar_type_t<Type>;
     auto const d_haystack = column_device_view::create(haystack, stream);
-    auto const s          = static_cast<ScalarType const*>(&needle);
+
+    // `get_scalar_device_view` only accepts non-const reference so we need to strip const.
+    auto const s        = static_cast<ScalarType const*>(&needle);
+    auto const d_needle = get_scalar_device_view(const_cast<ScalarType&>(*s));
```

Comment:
That's unfortunate that we have to `const_cast` here. I think this indicates a problem that we aren't distinguishing mutable/immutable views, but solving that problem here is out of scope. However, we can probably make this casting a little cleaner -- there's no need to use a pointer in the middle. See example: https://github.com/rapidsai/cudf/blob/bc5e769cdd3c974b4fa9a38724ee479d7febcb8e/cpp/include/cudf/detail/calendrical_month_sequence.cuh#L46-L47

```suggestion
    // `get_scalar_device_view` only accepts non-const reference so we need to strip const.
    auto const d_needle = get_scalar_device_view(static_cast<cudf::scalar_type_t<Type>&>(const_cast<scalar&>(needle)));
```

Then delete `ScalarType` above because it's only needed once.

Reply:
The `scalar/const` issue has been around for awhile: https://github.com/rapidsai/cudf/issues/4365